### ISSUE TITLE
Pictograms both directions on oneway street

### ISSUE THREAD

**Issue by frankie2784** [streetcomplete/StreetComplete#4721]
There are a number of oneway roads in my area with explicit sharrows in both directions, see below:

![image](https://user-images.githubusercontent.com/19356194/210486480-1f9292db-4889-40ca-9a85-8dc3bc14dcbc.png)

Unfortunately, SC doesn't provide the option to select sharrows in the opposite direction, only "none, but cyclists may use road in both directions" - which tags the way with cycleway:right=no + oneway:bicycle=no

Is it possible to add the sharrow option here, so that in the above case we would get:

cycleway:both=shared_lane
cycleway:both:lane=pictogram
cycleway:right:oneway=-1
oneway:bicycle=no

?

**Comment by westnordost** [comment]
Sharrows in opposite direction of a oneway doesn't make sense. `shared_lane` implies that the space is shared with cars (etc.) but there are no cars in contra-flow.

The _"none, but cyclists may use road in both directions"_ selection is specifically made for this case.

**Comment by frankie2784** [comment]
Interesting perspective. Yet here we are with an example of sharrows in the
contra-flow direction. Seems to me it would be useful to distinguish
between signage indicating bikes can use a particular road and actual
pictograms (not just in this case, but other instances of shared roads I
have seen with no sharrows but signs to "watch for bikes" or with bike
symbols). The reason I think this is important is that sharrows provide a
visual cue along the entire length of the road to a motorist to watch for
bikes, as distinct from the more common "cyclists permitted" sign which is
basically just an indicator of the legality of contra-flow travel for bikes.

Maybe that's a broader discussion for OSM than SC, though.

**Comment by westnordost** [comment]
Well,  I guess simply a sign "no entry, bicycles allowed" can make a difference from additionally pictograms painted through the whole road, reminding drivers that bicycle contraflow is to be expected.

@matkoniecz what do you think?  I think you were the first one to notice such pictograms but suggested to tag such situations as "no" plus not-a-oneway for bicycles.

**Comment by matkoniecz** [comment]
For start: I never bothered with mapping sharrows itself for multiple reasons

- they are approximately useless as bicycle infrastructure and situation is basically unchanged by their existence
- it is tricky to decide where they start and end - for example road has single sharrow marking at start and at the end: which part should be marked as having sharrows? For short street it can be argued that entire is marked - but how long it needs to be to mark middle part as different?
- And really, splitting road in three over some painted sharrows is a bit too much - even for me and I split roads to mark 20m lane of contraflow near crossing (that later turns into contraflow without dedicated lane)
- at least in my city they are not always repainted making marking them even more drudgery (due to snow/sand/salt/freezing/refreezing in Poland basically all painted markings need to be repainted each year, at most every few years - unless thermoplastic road marking paint are used but these are high enough to be irritating for cyclists and especially bad for sharrow markings)
- there are some claims that they will be noticed by drivers and make them more aware of cyclists - I am quite sceptical that drivers that need this reminder actually notice sharrows

No matter whether sharrows are painted or not on contraflow without lane I marked them with just  `oneway:bicycle=no` in manual mapping. I also used `cycleway=opposite` to indicate lack of dedicated lane but people convinced me that it is not needed.

**Comment by frankie2784** [comment]
I agree with many of your points Mateusz. I'm finding it is somewhat
subjective what parts of a road are considered shared. Sometimes there is a
very obvious chain of sharrows indicating a route, other times they are few
and far between. I find the similar issues with tagging regular bike lanes.
For example, how faded does a line need to be for it to no longer be
considered an exclusive lane?

I personally don't think that's a reason to not tag these features as they
are still part of the infrastructure, and there on the road for a reason -
it just means the tagging may be a bit fuzzy at the edges in terms of
interpretation.

In terms of uses, appropriately sized and spaced sharrows definitely have
an impact on driver expectations and behaviour. They can also be used to
identify a cycling corridor. So understanding where these sharrows exist is
still important for journey planning applications.

The alternative is to simply mark a series of roads in a relation as a
cycling route, which should work in many cases, but that still doesn't tell
you about the specific conditions on the road.

**Comment by westnordost** [comment]
Uh well, let's not invent new tagging in this issue tracker.

Mateusz, in Krakow, do all not-a-oneway for cyclists roads have pictograms / arrows as markings on the ground, or only some? Is there any regularity to it where there are markings and where there are no markings?

**Comment by matkoniecz** [comment]
> Mateusz, in Krakow, do all not-a-oneway for cyclists roads have pictograms / arrows as markings on the ground, or only some? Is there any regularity to it where there are markings and where there are no markings?

Some have, some do not have. It seems to be primarily function of which they bother to repaint and this seems random but maybe there is some logic behind this.

It seems that newly created contraflows are more linked to be painted (maybe because they paint nearly all and repaint only some)?

There are also some analogues like this signs reminding about legally required separation https://krakow.naszemiasto.pl/krakow-stanely-znaki-ktore-przypominaja-kierowcom-o/ga/c1-8286122/zd/62106552

![screen01](https://user-images.githubusercontent.com/899988/210808565-5af64c26-1a76-495a-acda-a29c8a8907ad.png)

**Comment by westnordost** [comment]
Hmm, so following this line of argumentation, do ("real") sharrows not have the same issue as bicycle-in-contraflow markings? that they get washed out after some time? To me, this does not sound like a valid reason not to map sharrows at all.

**Comment by matkoniecz** [comment]
In this case I map that cycling is legal in both directions which is a real difference - and I would mark contraflow without lane that has sharrows in the same way as contraflow without lane that has no sharrows.

**Comment by westnordost** [comment]
No,  sorry, I meant normal streets with sharrows, not oneways

**Comment by matkoniecz** [comment]
And main difference is that in normal editing I was not marking "cycleway:both=no" either... So SC is in worse situation here...

**Comment by westnordost** [comment]
I don't follow. 

What are you inputting for sharrows, both ways,  then?

**Comment by westnordost** [comment]
My point is the following:

1. I agree with @frankie2784 that it can make a difference whether there is just a sign at the entrance to the oneway about bicycle contraflow or if there are pictograms on the road in regular intervals. I'd even go so far that difference between `no` and `shared_lane` is more significant on oneways than for non-oneways as drivers are not only reminded that there can be bicycle traffic at all, but also that it may be in contraflow and they will tend to drive in a way as to try not to drive over the bicycle pictograms in contraflow (I know I would).
2. The street markings vanish with time either way. I.e. they vanish as much on oneway roads as on non-oneway roads. The fact that they vanish over time is not a good argument to map them at all.
3. Whether barely visible markings or markings that are very far apart should count as sharrows or not is up to each surveyor to decide. If you decide in your town that they are not visible (anymore) to make a difference or too far apart to be "real" sharrows, you can tag them as `no`. However, that should not bar other users to explicitly mark them as sharrows if they are clearly sharrows.

**Comment by westnordost** [comment]
So what I did is to remove the removal of "lane shared explicitly with other traffic" from the selectable options in a oneway but leave the rest the same.
